squeal-postgresql/src/Squeal/PostgreSQL/Expression/Time.hs

```diff
@@ -126,6 +130,46 @@ makeTimestamptz ::
    , null 'PGint4, null 'PGint4, null 'PGfloat8 ] ---> null 'PGtimestamptz
 makeTimestamptz = unsafeFunctionN "make_timestamptz"
 
+{-|
+Truncate a timestamp with the specified precision
+
+>>> printSQL $ dateTrunc ("day" *: makeTimestamp (2010 :* 5 :* 6 :* 14 :* 45 *: 11.4))
+date_trunc((E'day' :: text), make_timestamp((2010 :: int4), (5 :: int4), (6 :: int4), (14 :: int4), (45 :: int4), (11.4 :: float8)))
+-}
+dateTrunc ::
+  '[ null 'PGtext, null 'PGtimestamp ] ---> null 'PGtimestamp
+dateTrunc = unsafeFunctionN "date_trunc"
```

Comment:
From the [documentation for `date_trunc`](https://www.postgresql.org/docs/9.1/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC), the first argument has only valid values of:
```
microseconds
milliseconds
second
minute
hour
day
week
month
quarter
year
decade
century
millennium
```
This should be covered by Squeal's `TimeUnit` type (although it appears that `quarter` is missing). Also we may make it as polymorphic in Haskell as it is in Postgres.
```
dateTrunc :: time `In` '[ 'PGtimestamp, 'PGtimestamptz] => TimeUnit -> null time --> null time
```

Reply:
Sounds good!